systems/controllers/constructQPDataPointerMex.cpp

```diff
@@ -338,7 +338,20 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 
   // terrain_map_ptr
   if (!mxIsNumeric(prhs[narg]) || mxGetNumberOfElements(prhs[narg])!=1) mexErrMsgIdAndTxt("Drake:constructQPDataPointerMex:BadInputs","this should be a map pointer");
-  memcpy(&pdata->map_ptr,mxGetPrSafe(prhs[narg]),sizeof(pdata->map_ptr));
```

Comment:
mxGetPrSafe actually caught this one by throwing an error when used in DRC (because our map_ptr is not, in fact, a double)